### DISCUSSION TITLE
Run another script when pyinstaller is complete?

### DISCUSSION THREAD

**Discussion by kgroening05** [brentvollebregt/auto-py-to-exe#368] [Ideas]
I'm using this GUI to convert a bunch of smaller script files. After converting each file, I'm manually running a Powershell script to move files from the /output directory to other places as needed. I was thinking it would be convenient if there was an option to automatically run a script after pyinstaller finishes.

I was wanting to make a contribution to this repo and perhaps add this as a feature. My thought is I could basically copy the "Manually Provide Options" input under Settings to create another input field that would append additional commands after pyinstaller commands.

An example output command might then look something like:
```
pyinstaller --noconfirm --onedir --console  "C:/example.py" ; somescript.ps1 -option
```

where the ` ; somescript.ps1 -option` was added in the new input field.

Would adding something like this make sense?

**Comment by brentvollebregt** [comment]
I get what you're saying, just it may be easier to the `pyinstaller` command directly in your workflow. 

So instead of opening auto-py-to-exe every time and then running a script to move the files, you could use the generated `pyinstaller` command (shown at the bottom of the UI) at the beginning of `somescript.ps1`.

I suppose this will only work if you know the options beforehand.

**Reply by kgroening05** [reply]
Yeah, I thought about doing something like that too. I did put together a script to use `pyinstaller` to convert all 20+ of the python scripts, but I think that the people who will be maintaining the scripts I'm writing are more comfortable with a GUI tool than with the command line. 

I think `auto-py-to-exe` would be for converting one or a few scripts at a time since if they make changes they will have to convert them to EXE to test them. So potentially they're creating the same script many times while debugging. Using the GUI is pretty convenient for iterating through that process for a couple scripts.

**Reply by brentvollebregt** [reply]
Sorry, just trying to get a use case here - I don't quite understand how a script will not make this a lot faster to do.

> So potentially they're creating the same script many times while debugging. Using the GUI is pretty convenient for iterating through that process for a couple scripts.

If it's the same scripts every time, then the same options would be getting applied (I would think) - using one script could convert all 20+ in a single run and be a significant amount faster than configuring the UI for 20 times.

**Reply by kgroening05** [reply]
No worries, maybe I'm thinking about this wrong. 

To me, the purpose of this tool is to make `pyinstaller` more accessible and easier to use for people who aren't as comfortable digging through documentation and using the command line. 

You're right that the same options would be getting applied for each script, but the thing that might be changing between runs is the source script path. Since the other options stay in the UI between runs, it's pretty easy to convert one or two scripts by changing the file in the UI vs changing the path in a script. 

For when all 20+ of the scripts need to be converted I have a file `generate-all-exe-files.ps1` for that purpose, but when only one or two scripts need to be modified it seems like the UI is an easier workflow.

I could definitely be misunderstanding how much work would be involved in adding a feature like this, but it seems like a fun little project to implement.
